Jormungandr: activate autocomplete on instance

antoine-de opened the pull request:
add the ability to use the global autocomplete (mimir) instead of the
instance's autocomplete
(ie `/v1/coverage/<instance>/places `uses mimir)

By default it's the instance autocomplete that is used.

To activate it add to the configuration:
```json
"autocomplete": {
        "class":
"jormungandr.autocomplete.geocodejson.GeocodeJson",
        "args" : {
            "host": "the host"
        }
    },
```

Note: I don't like that we use the v1's marshall in the autocomplete,
but it's what was done for kraken and I don't see a nice way to avoid
this

Note2:
I changed the way the autocomplete was given in the configuration (to be consistent with the realtimeproxies), thus we'll need to change the jormungandr configuration (but it's not used yet, so no breakdown)

Note 3: lots of files changed because of some refactors for the tests

antoine-de committed:
Jormungandr: activate autocomplete on instance

add the ability to use the global autocomplete (mimir) instead of the
instance's autocomplete
(ie /v1/coverage/<instance>/places uses mimir)

By defualt it's the instance autocomplete that is used.

To activate it add :
```json
"autocomplete": {
        "class_path":
"jormungandr.autocomplete.geocodejson.GeocodeJson",
        "kwargs" : {
            "host": "the host"
        }
    },
```
to the configuration

Note: I don't like that we use the v1's marshall in the autocomplete,
but it's what was done for kraken and I don't see a nice way to avoid
this

pbougue reviewed (commented):
Maybe add to test's conf the use of kraken in autocomplete, just to test conf loading (cause result would obviously be the same).

As for the conf naming it would be nice to have something consistent (and retrocompatible) :+1: 
I have no opinion on what's the best naming though.

antoine-de committed:
remove old bragi retrocompatibility

antoine-de committed:
refactor georef

the way the 'way'were build was buggy, we used a baddly implemented copy
constructor.

antoine-de committed:
change autocomplete params

for coherence purpose, we now use the same way to build all dynamic
python objects:

```
"autocomplete": {
        "class":
"jormungandr.autocomplete.geocodejson.GeocodeJson",
        "args" : {
            "host": "the host"
        }
    },
```

we thus need to change the jormungandr configuration

antoine-de committed:
change instances configuration in integration tests

we now use a generic instance configuration to be able to pass all the
parameters we want

antoine-de committed:
move utils method and add check

better check of autocomplete response

antoine-de committed:
add an integration test on bragi

antoine-de commented:
I updated the PR with some tests.

It impacts way more file due to some refactoring needed along the way..

antoine-de committed:
corrections after review

antoine-de committed:
add newline
